### PR TITLE
Mice in maints as ghost spawners

### PR DESCRIPTION
**Pull request by OcelotTheOcelot**
## About The Pull Request
This PR adds every mouse spawned in maintenance corridors to the ghost spawners menu. Mice are unable to communicate via speech until they gain sentience. Player-controlled mice now also have the ability to chew power cabels and die because of that as well as the original simplemobs used to.

## Why It's Good For The Game
Ghost players now gain the ability to squeak at the crew, to gnaw power cables and to organize a march into the SM crystal.

## Changelog
:cl:
add: maint mice ghost spawners
code: minor changes in mouse.dm
/:cl:

### CODE REVIEW

**Comment by TotallyNotChief** [comment]
yes we
we are mice now

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 1]
```diff
@@ -42,6 +45,61 @@
...
+	return "Do your business with the cheese and squeaking." //a mild reference to the hitchhiker's guide to the galaxy
+
+/mob/living/simple_animal/mouse/get_spawner_flavour_text()
+	return "You are a maintenance mouse. Cooperate with your fellow mice to go on a holy crusade for some cheese or just get toasted by gnawing the first cable you find."
+
+/mob/living/simple_animal/mouse/attack_ghost(mob/user)
+	. = ..()
+	if(key || stat || !MOUSE_GHOST_ROLE_ENABLED)
+		return
+	var/mouse_ask = alert("Do you really want to become a mouse?", "Become a mouse?", "Yes", "No")
```
Comment (TotallyNotChief) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:84]:
```suggestion
	var/mouse_ask = alert("Do you really want to become a mouse?", "Become a mouse?", "Yes", "Yes")
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 2]
```diff
@@ -42,6 +45,61 @@
...
+	if(mouse_ask == "No" || QDELETED(src))
+		return
```
Comment (TotallyNotChief) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:86]:
```suggestion

```

**Review by TotallyNotChief — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 2 above)
Comment (TotallyNotChief) [reply code/modules/mob/living/simple_animal/friendly/mouse.dm:86]: no turning back

**Review by ro5490 — APPROVED** [review]
Looks pretty good, nice work

**Comment by Moccha-Bee** [comment]
really based

**Comment by Kryyto** [comment]
Work well, There is no description in the spawner menu though 
image

**Comment by OcelotTheOcelot** [comment]
> Work well, There is no description in the spawner menu though
> image

I don't know why doesn't it appear, I've overriden the 'get_spawner_desc' and 'get_spawner_flavour_text' procs tho. I will try to fix this in me next PR where I will also fix that SCP-173 issue.

**Comment by OcelotTheOcelot** [comment]
There was a `isllime(O)` check in the spawners menu that only used `get_spawner_desc` when the object is a slime. As I can see, `get_spawner_flavour_text` returns atom's desc by default, therefore the `isslime(O)` check is redundant.
In case the spawner wasn't a slime, the `short_desc` field would stay an empty string, why not use `get_spawner_desc` instead, which returns atom's name by default?

**Comment by Kryyto** [comment]
> 
> 
> There was a `isllime(O)` check in the spawners menu that only used `get_spawner_desc` when the object is a slime. As I can see, `get_spawner_flavour_text` returns atom's desc by default, therefore the `isslime(O)` check is redundant.
> In case the spawner wasn't a slime, the `short_desc` field would stay an empty string, why not use `get_spawner_desc` instead, which returns atom's name by default?

Honestly, I have no idea why ```isslime``` is a thing since it should work in any case, I guess it was made due to a bug that was fixed later on, and the guys forgot to remove it ?

**Pull request merged**